docs/deprecation.md

```diff
@@ -0,0 +1,28 @@
+# Deprecation Policy & Breaking Changes
+
+This documentation gives a short outline of our deprecation strategy in the
+project.
+
+## User-facing commands and their arguments
+
+When we rename a command or make a previously optional argument required,
+we usually try to keep the old command invocations working for 6
+months (so 6 releases, since we release monthly) with a deprecation message.
+The message should inform the user that the previous workflow is deprecated
+and to be removed in the future.
+
+## Niche commands
+
+For commands with a niche user audience or something we assume is rarely used
+(we sadly have no data), we take the liberty to remove the old behavior within
+two releases. This means that you can change the old command to immediately
+return a error. A example is if we want to rename `jj debug reindex` to
+`jj util reindex` in a release, then we make `jj debug reindex` an error in the
+same patchset.
```

Comment:
Does this mean that even for niche commands, it's not okay to make breaking changes from one release to the next? I'm personally fine with making changes like #3983 in the future too.

Reply:
> Does this mean that even for niche commands, it's not okay to make breaking changes from one release to the next?

I don't think so, as it currently only mentions renames or removals not extending basic functionality.